### ISSUE TITLE
adasum do not support train_mixed_precision

### ISSUE THREAD

**Issue by guangdongliang** [horovod/horovod#3429]
In [link](https://github.com/horovod/horovod/blob/master/examples/pytorch/pytorch_mnist.py), adasum optimizer do not support method skip_synchronize and synchronize when train with mixed precision. without that methods loss will all become nan.

**Comment by schmidt-jake** [comment]
Mixed precision training is common / standard these days — without mixed precision support for Adasum, it feels like Adasum is a nonstarter.
